### PR TITLE
[MAINTENANCE] Update `Domain` computation in MetricConfiguration

### PR DESCRIPTION
**Pull request by alexsherstinsky**
### Scope
* We insure that only relevant `Domain` arguments are present (i.e., column names).
* Revert `ParameterNode` conversion to using `isinstance()` for type comparisons.
* Update unit tests.

Please annotate your PR title to describe what the PR does, then give a brief bulleted description of your PR below. PR titles should begin with [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], or [CONTRIB]. If a new feature introduces breaking changes for the Great Expectations API or configuration files, please also add [BREAKING]. You can read about the tags in our [contributor checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist).

Changes proposed in this pull request:
- JIRA: GREAT-1347/GREAT-181/GREAT-1432
-
-

After submitting your PR, CI checks will run and @cla-bot will check for your CLA signature.

For a PR with nontrivial changes, we review with both design-centric and code-centric lenses.

In a design review, we aim to ensure that the PR is consistent with our relationship to the open source community, with our software architecture and abstractions, and with our users' needs and expectations. That review often starts well before a PR, for example in GitHub issues or Slack, so please link to relevant conversations in notes below to help reviewers understand and approve your PR more quickly (e.g. `closes #123`).

Previous Design Review notes:
-
-
-

### Definition of Done
Please delete options that are not relevant.

- [x] My code follows the Great Expectations [style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style)
- [x] I have performed a [self-review](https://docs.greatexpectations.io/docs/contributing/contributing_checklist) of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added [unit tests](https://docs.greatexpectations.io/docs/contributing/contributing_test#writing-unit-and-integration-tests) where applicable and made sure that new and existing tests are passing.
- [x] I have run any local integration tests and made sure that nothing is broken.

Thank you for submitting!

### CODE REVIEW

File: tests/validator/test_metric_configuration.py [hunk 1]
```diff
@@ -55,14 +55,11 @@ def test_metric_configuration_get_domain(
...
-        domain_kwargs={
-            "batch_id": "abc123",
-        },
+        domain_kwargs={},
     )
     assert column_histogram_metric_config.get_domain() == Domain(
         domain_type=MetricDomainTypes.COLUMN,
         domain_kwargs={
             "column": "my_column",
-            "batch_id": "def456",
```
Comment (NathanFarmer) [inline tests/validator/test_metric_configuration.py:66]: Why did we drop `batch_id`?

Comment (alexsherstinsky) [reply tests/validator/test_metric_configuration.py:66]: @NathanFarmer It belongs in `metric_configuration` domain -- but not in the metric `Domain` as in the proper `Domain` object.

File: great_expectations/validator/metric_configuration.py [hunk 2]
```diff
@@ -69,10 +69,40 @@ def metric_dependencies(self, metric_dependencies) -> None:
 
     def get_domain(self) -> Domain:
         """Return "Domain" object, constructed from this "MetricConfiguration" object."""
-        return Domain(
-            domain_type=self.get_domain_type(),
-            domain_kwargs=self._metric_domain_kwargs,
-        )
+        domain_type: MetricDomainTypes = self.get_domain_type()
+
+        if domain_type == MetricDomainTypes.TABLE:
```
Comment (cdkini) [inline great_expectations/validator/metric_configuration.py:74]: think this should be `is`

Comment (alexsherstinsky) [reply great_expectations/validator/metric_configuration.py:74]: @cdkini Agreed to handle broadly in very near-term follow-up work.

**Review by NathanFarmer — APPROVED** [review]
LGTM!

**Pull request merged**